src/sidebar/components/ThreadList.tsx

```diff
@@ -33,6 +34,45 @@ export type ThreadListProps = {
   threads: Thread[];
 };
 
+/**
+ * Find the selector identifying the document segment which an annotation
+ * belongs to.
+ */
+function getSegmentSelector(ann: Annotation): EPUBContentSelector | undefined {
+  return ann.target[0].selector?.find(s => s.type === 'EPUBContentSelector') as
+    | EPUBContentSelector
+    | undefined;
+}
```

Comment:
This function specifically feels like it might belong in, say, `annotation-metadata` or another helper module?

Reply:
I expect it will make sense to extract this into a separate module in future.